### ISSUE TITLE
Initial price check is wrong in `create_auction`

### ISSUE THREAD

**Issue by matt-user** [FuelLabs/sway-applications#356]
### Describe the bug

The english auction is incorrectly checking the initial price when creating auctions.

### Steps to reproduce

The auctions should be able to sell an nft and be bid on by a token asset in whatever increment of the token asset the seller of the nft wants.  However, this is not the case as shown in the code below.

### Minimal reproducible code example

```Rust
match bid_asset {
    AuctionAsset::TokenAsset(asset) => {
        require(asset.amount() == 0, InitError::BidAssetAmountNotZero);
    },
    AuctionAsset::NFTAsset(asset) => {
        require(asset.token_id() == 0, InitError::BidAssetAmountNotZero);
    }
}

// Ensure that the `sell_asset` struct and what was sent in the transaction match
match sell_asset {
    AuctionAsset::TokenAsset(asset) => {
                // Selling tokens
                // TODO: Move this outside the match statement when StorageVec in structs is supported
                // This issue can be tracked here: https://github.com/FuelLabs/sway/issues/2465
this should be moved==>require(initial_price != 0, InitError::InitialPriceCannotBeZero);<=====
                require(msg_amount() == asset.amount(), InputError::IncorrectAmountProvided);
                require(msg_asset_id() == asset.asset_id(), InputError::IncorrectAssetProvided);
            },
    AuctionAsset::NFTAsset(asset) => {
        // Selling NFTs
        let sender = msg_sender().unwrap();
        // TODO: Remove this when StorageVec in structs is supported
        // This issue can be tracked here: https://github.com/FuelLabs/sway/issues/2465
=======>require(initial_price == 1, InitError::CannotAcceptMoreThanOneNFT);<========= this should be moved
        transfer_nft(asset, Identity::ContractId(contract_id()));
    }
}
```

### Rust Version

1.65.0

### Forc Version

0.31.1

### Expected behavior

Users should be able to create auctions selling NFTs with any initial price (for tokens).

### Additional context

Basically the initial price checks are under the `sell_asset` match statement and they should be under the `bid_asset` match statement

**Comment by matt-user** [comment]
closed [here](https://github.com/FuelLabs/sway-applications/pull/360)
